File: backend-junglehunter/src/main/java/at/htl/junglehunter/entity/Trail.java
package at.htl.junglehunter.entity;

import at.htl.junglehunter.dto.TrailDto;
import io.quarkus.hibernate.orm.panache.PanacheEntity;

import javax.persistence.*;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

@Entity
public class Trail extends PanacheEntity {

    @Column(length = 100)
    public String name;

    public Double length;

    public Boolean hasGpx = false;

    @ManyToOne
    public Route route;

    @OneToMany(mappedBy = "trail", cascade = CascadeType.REMOVE)
    public List<ControlPoint> controlPoints;

    public Trail(String name, double length) {
        this.name = name;
        this.length = length;
    }

    public Trail() {
    }

    public void update(TrailDto trail) {
        this.name = trail.getName();
        this.length = trail.getLength();
    }

    public static List<TrailDto> getDtos(Stream<Trail> entityStream) {
        return entityStream
                .map(TrailDto::map)
                .collect(Collectors.toList());
    }
}

File: ios-app-junglehunter/jungle-hunter-app/Trail.swift
class Trail {
    var name: String
    var length: Double
    var controlpoints: [ControlPoint]
    
    init (trailDict: [String:Any]) {
        self.name = trailDict["name"] as! String
        self.length = trailDict["length"] as! Double
        self.controlpoints = (trailDict["controlPoints"] as! [[String:Any]]).map{ ControlPoint(controlpointDict: $0) }
    }
    
    init (name: String, length: Double, controlpoints: [ControlPoint]) {
        self.name = name
        self.length = length
        self.controlpoints = controlpoints
     }
}
